game_plugins/hase_und_igel_2018/shared/sc/plugin2018/GameState.java

```diff
@@ -232,160 +219,145 @@ public Player getOtherPlayer() {
-        && (index != Constants.NUM_FIELDS - 1);
+        && (index != Constants.LAST_FIELD_INDEX);
   }
-
+  
   /**
-   * Überprüft ob der angegebene Spieler an erster Stelle ist. Wenn sich beide
+   * Überprüft, ob der angegebene Spieler an erster Stelle ist. Wenn sich beide
    * Spieler im Ziel befinden wird zusätzlich überprüft, ob <code>player</code>
    * weniger Karotten besitzt als der Gegenspieler.
-   * 
+   *
    * @param player überprüfter Spieler
+   *
    * @return true, falls Spieler an erster Stelle
    */
-  public final boolean isFirst(final Player player)
-  {
+  public final boolean isFirst(final Player player) {
     Player o = this.getOpponent(player);
     boolean isFirst = o.getFieldIndex() <= player.getFieldIndex();
     if (player.inGoal() && o.getFieldIndex() == player.getFieldIndex())
       isFirst = isFirst
           && player.getCarrots() < o.getCarrots();
     return isFirst;
   }
-
+  
   /**
    * Gibt den Feldtypen an einem bestimmten Index zurück. Liegt der
    * gewählte Index vor dem Startpunkt oder hinter dem Ziel, so wird
-   * <code>INVALID</code> zurückgegeben.
+   * {@link FieldType#INVALID} zurückgegeben.
    *
    * @param index die Index auf der Rennstrecke
-   * @return Feldtyp an Index
+   *
+   * @return Feldtyp an index
    */
-  public final FieldType getTypeAt(final int index)
-  {
+  public final FieldType getTypeAt(final int index) {
     return board.getTypeAt(index);
   }
-
+  
   /**
-   * Findet das nächste Spielfeld vom Typ <code>type</code> beginnend an
-   * Index <code>index</code> auf diesem Spielbrett.
+   * Findet das nächste Spielfeld vom Typ <code>type</code> beginnend bei
+   * Feldnummer <code>index</code> auf diesem Spielbrett.
    *
-   * @param type Feldtyp
+   * @param type  Feldtyp
    * @param index Index
+   *
    * @return Index des nächsten Feldes genannten Typs
    */
-  public final int getNextFieldByType(FieldType type, int index)
-  {
+  public final int getNextFieldByType(FieldType type, int index) {
     return this.board.getNextFieldByType(type, index);
   }
-
+  
   /**
    * Findet das vorherige Spielfeld vom Typ <code>type</code> beginnend an Index
    * <code>index</code> auf diesem Spielbrett.
-   * @param type Feldtyp
+   *
+   * @param type  Feldtyp
    * @param index Index
+   *
    * @return Index des vorherigen Feldes genannten Typs
    */
-  public final int getPreviousFieldByType(FieldType type, int index)
-  {
+  public final int getPreviousFieldByType(FieldType type, int index) {
     return this.board.getPreviousFieldByType(type, index);
   }
-
+  
   /**
    * liefert die aktuelle Zugzahl
+   *
    * @return Nummer des aktuellen Zuges (Zaehlung beginnt mit 0)
    */
   public int getTurn() {
     return turn;
   }
-
+  
   /**
    * Setzt die aktuelle Zugzahl. Nur für den Server relevant
+   *
    * @param turn neue Zugzahl
    */
-  public void setTurn(int turn) throws InvalidGameStateException {
```

Comment:
Tatsaechlich ist das Problem schon in diesem PR...

Reply:
jap, der testclient-PR baut ja auf diesem hier auf.